portfolio/src/main/webapp/script.js

```diff
@@ -12,39 +12,70 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+/** @param slideIndex slide currently being displayed */
 var slideIndex = 1;
+
+/**
+ * Supported class states for dots and slides.
+ * @enum {string}
+ */
+const states = {
+  SHOW_DOT: ' active',
+  HIDE_DOT: '',
+  SHOW_SLIDE: 'block',
+  HIDE_SLIDE: 'none',
+}
+
 showSlides(slideIndex);
 
-// Next and previous controls
+/**
+ * Function computes which slide should be shown and class showSlides on that
+ * specific slide. Function is called when either arrows are pressed.
+ * @param {integer} n where n = 1 to go forward or n = -1 to go back
+ */
 function plusSlides(n) {
   showSlides(slideIndex += n);
 }
 
-// Allows changing slides through the dots
+/**
+ * Function calls showSlides on the slide that corresponds to the dot pressed.
```

Comment:
nit: this comment can cause confusion because the user does not know what showSlides means. I would rephrase to something like:

Allows changing of slides through the dots.

@param (integer} n where n is the number of the dot that is pressed